addons/website_blog/static/src/js/wysiwyg_adapter.js

```diff
@@ -78,3 +117,34 @@ patch(WysiwygAdapterComponent.prototype, {
         }
     },
 });
+
+export class WebsiteBlogSnippetsMenu extends snippetsEditor.SnippetsMenu {
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * @override
+     */
+    _computeSnippetTemplates(html) {
+        this._super(...arguments);
+        const postContentEl = this.options.document.getElementById('o_wblog_post_content');
+        if (postContentEl) {
+            // Patch all droppable snippet templates.
+            const usesRegularCover = this.$body[0].querySelector('.o_wblog_regular_cover_container');
+            const targetClass = usesRegularCover ? 'container' : 'o_container_small';
+            const removedClass = usesRegularCover ? 'o_container_small' : 'container';
+            for (const snippetEl of this.$snippets) {
+                snippetEl.querySelectorAll([
+                    `section .${removedClass}`,
+                    'section .container-fluid',
+                ]).forEach(el => {
+                    el.classList.remove('container-fluid', removedClass);
+                    el.classList.add(targetClass);
+                });
+            }
+        }
+    }
+}
```

Comment:
Note that I wasn't able to open the runbot do an issue 
![image](https://github.com/user-attachments/assets/72481d0d-e16e-470e-b005-7919692d7e06)

We had a quick discussion with @bso-odoo and it's all good to me. By default, we'll use the "small" content-width (or the one the client used, i.e. full or small). Users could change value for each snippet they used if they want.

In the old branch (i.e. master-refine_blog_content-ome), I see an issue when we add the sidebar in a blog post, we cannot change the content-width to full anymore. We should check in this new branch.

Reply:
> By default, we'll use the "small" content-width (or the one the client used, i.e. full or small). 

@lebl-odoo Francesco (cc @fres-sudo) is the one finishing up this task. I am not sure what you mean by "the one the client used". The question we had for you (which I was hoping for FRCA to ask you IRL yesterday's afternoon) was:

There are two possibilities. In either case, when you create a blog post, we will create the block with "the right size", meaning small by default or "container" if "regular cover" is enabled. The question here is for new snippets which are added *afterwards* in the content:
- Either we try to make them match the "right size" by default when they are dropped
    - Advantage: we try to follow the size of the initial content automatically
    - Disadvantage: it's more code, snippets are different between blog and normal pages and potentially, the "small" size does not make sense in some snippets
- Or, we do nothing automatically and the snippets are the same in blog post than anywhere else

I am not that sure that the first option is that good anymore. I mean, if you drop the "items" snippet for example, you would get this:
![image](https://github.com/user-attachments/assets/ddd9ebf4-8b93-430f-8384-eb19158e2eed)
Sure, the user can fix it... but maybe we better leave the snippet be as they were designed in the first place.

Or maybe we could do an in-between and explicitly list the snippets we want to be automatically adapted... but it's more code to maintain.

Reply:
I'm sorry for the runbot that do not runs, I'm aware of the issue. I was waiting to you to come in GR3 to discuss in person about the thing that @qsm-odoo just said (which I agree with).

I checked this morning the issue with @SergeBayet, and there are some strange things going on that needs more investigation, for that reason I was waiting for your feedback before proceeding with the fix, since the part that is not working is the part responsible to handle the "small" content by default.

Reply:
> by "the one the client used"

BSO tolds me - if I understand well - the migration script will check which mode was enabled (either normal or "increase readability") and according to this value, we'll move the existing content into a new "text" snippet with the according content-width (either "regular" or "small").

Regarding the rest of your concern, the way we read blog post has changed over the past years. Now, it is expected from users to have small content-width (see how medium displays the content). Especially now with wide-large screens.

Here is what we'll get by using the "regular" content-width in a new blog post:
![image](https://github.com/user-attachments/assets/88e6d3c0-054c-4203-a655-37a933506e54)

Note that when we enable the sidebar, it works better.

However, I heard your arguments. Since users used to have the regular size when they drop a snippet, applying an unexpected size could lead to misunderstanding and degrade the UX. This argument could take over on what I said previously. In that case, I would like to be able to change the content-width of the breadcrumb, tags and comment section (see previous screenshot, all the small boxes).

Reply:
@lebl-odoo cc @fres-sudo 

> Note that when we enable the sidebar, it works better.

Strange you think that, I have the exact opposite opinion: I think it is still awful for most snippets when you do use the sidebar... but unfortunately we can't do much about that, using snippets in that sidebar configuration would just be not recommended I guess.

> Here is what we'll get by using the "regular" content-width in a new blog post:

Yes, but the user can then resize the content as he is used too everywhere else or... he can just not drop a text block like you did: there is no point, there is already a "Start writing here..." section. But if we want, like I said, we could also do the solution for specific snippets, maybe only the text one... who knows?

> This argument could take over on what I said previously.

So, solution n°2 and we leave **all** drag&drop snippets untouched?

> In that case, I would like to be able to change the content-width of the breadcrumb, tags and comment section (see previous screenshot, all the small boxes).

Right now there is an automatic system: those follow the size of the first snippet in the blog content automatically. You would prefer for the user to control those explicitly and one by one for each post? That would require new fields on the blog.post model and basically many changes in this task.

Reply:
@qsm-odoo 

> Strange you think that

Arf, my bad, we don't talk about the same thing most probably. Indeed, adding snippets in the sidebar doesn't make any sense. I meant the gap between "small" and "regular" content-width is lower when the sidebar is enabled

![image](https://github.com/user-attachments/assets/ccd946e4-0510-46ba-98a3-3e36a276300c)

vs

![image](https://github.com/user-attachments/assets/26d0aa20-b273-4b03-b258-85cf81d27058)

> So, solution n°2 and we leave all drag&drop snippets untouched?

Yes

> Right now there is an automatic system

Oh, didn't know. I need to play a bit with this feature first.

Reply:
> Indeed, adding snippets in the sidebar doesn't make any sense.

I was not speaking about that: just that when the sidebar is enabled, alignment is broken anyway and if you put a background color on a snippet, well... it's as bad as it could. The sizing discussion is not that important to me: of course the user can do ugly things.

> Yes [...] Oh, didn't know. I need to play a bit with this feature first.

Ok, so, @fres-sudo do as we said: remove that code (the old include you changed in a class which does nothing). We'll see the final result :+1: